horreum-backend/src/main/java/io/hyperfoil/tools/horreum/svc/RunServiceImpl.java

```diff
@@ -856,4 +858,52 @@ public Object updateSchema(Integer id, String path, String schemaUri) {
       em.flush();
       return schemas;
    }
+
+   private void onCalculateDataSets(String param) {
+      log.info("calculate_datasets");
+      String[] parts = param.split(";", 3);
+      if (parts.length < 3) {
+         log.errorf("Received notification to calculate dataset %s but cannot extract run ID.", param);
+         return;
+      }
+      int runId;
+      try {
+         runId = Integer.parseInt(parts[0]);
+      } catch (NumberFormatException e) {
+         log.errorf("Received notification to calculate dataset for run %s but cannot parse as run ID.", parts[0]);
+         return;
+      }
+      String token = parts[1].trim();
+      if (token.isBlank()) {
```

Comment:
Sorry if I missed this before, but are the checks for a non-blank token and role valid? I guess that it most cases you'd use the role-based authorization and the token *will* be blank. I guess that the test would fail - or my understanding might be incorrect.

Reply:
ok I'll remove the checks for blanks.